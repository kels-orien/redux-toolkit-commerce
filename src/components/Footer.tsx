import React from 'react'
import styled, { css } from 'styled-components'
import { lightGrey, lightGreySecondary } from '../utils/colors'
import { phone } from '../utils/media'
import { Link } from 'react-router-dom'


const Wrapper = styled.footer`
    width: 100%;
`

const SecondaryWrapper = styled.div`
    overflow: auto;
    text-align: center;
    background: ${lightGrey};
    border-bottom: 1px solid #eaeaea;
    border-top: 1px solid #eaeaea;
`

const TertiaryWrapper = styled.div`
    max-width: 500px;
    padding: 40px 20px;
    overflow: auto;
    margin: 0 auto;
    display: flex;

    ${phone(css`
    display: block;
  `)};
`
const InfoWrapper = styled.div`
    width: 33.3%;


    ${phone(css`
    width: 100%;
  `)};
`


const InfoHeader = styled.div`
    margin-bottom: .1rem;
`
const InfoSecondaryWrapper = styled.div`
    font-weight: 700;
    font-size: 1rem;
    margin-bottom: 10px;
`
const InfoItems = styled.div`
    font-weight: 400;
    color: ${lightGreySecondary};
`

const SmallPrint = styled.div`
    background: ${lightGrey};
    padding: 20px;
    text-align: center;
    color: #262626;
    font-size: 11px;
`

const SmallWrapper = styled.div`
    max-width: 1920px;
    margin: 0 auto;
    padding: 0 50px;
    ${phone(css`
    padding: 0 20px;
  `)};
`
const SmallText = styled.div`
    opacity: 0.7;
`


const Footer = () => {
    return (
        <Wrapper>
            <SecondaryWrapper>
                <TertiaryWrapper>

                    <InfoWrapper>
                        <InfoSecondaryWrapper>
                            <InfoHeader>
                                USEFUL LINKS
                            </InfoHeader>
                            <InfoItems>
                                <ul>
                                    <Link to={`/products/category/electronics`}><li>Electronics</li></Link>
                                    <Link to={`/products/category/jewelery`}><li>Jewelry</li></Link>
                                    <Link to={`/products/category/men's clothing`}><li>Men's Clothing</li></Link>
                                    <Link to={`/products/category/women's clothing`}><li>Women's Clothing</li></Link>
                                </ul>
                            </InfoItems>

                        </InfoSecondaryWrapper>

                    </InfoWrapper>
                    <InfoWrapper>
                        <InfoSecondaryWrapper>
                            <InfoHeader>HELP</InfoHeader>
                            <InfoItems>
                                <ul>
                                    <li><a href="/">FAQ</a></li>
                                    <li><a href="/">Shipping</a></li>
                                    <li><a href="/">Payment</a></li>
                                    <li><a href="/">Returns</a></li>
                                    <li><a href="/">Contact Us</a></li>
                                </ul>
                            </InfoItems>

                        </InfoSecondaryWrapper>
                    </InfoWrapper>
                    <InfoWrapper>
                        <InfoSecondaryWrapper>
                            <InfoHeader>SOCIAL MEDIA</InfoHeader>
                            <InfoItems>
                                <ul>
                                    <li><a href="/">Facebook</a></li>
                                    <li><a href="/">Instagram</a></li>
                                    <li><a href="/">Twitter</a></li>
                                </ul>
                            </InfoItems>

                        </InfoSecondaryWrapper>

                    </InfoWrapper>
                </TertiaryWrapper>
            </SecondaryWrapper>
            <SmallPrint><SmallText>Â© 2021 Korien. All Rights Reserved. |<a href="/"><b>Term of use</b></a> | <a href="/"><b>Privacy Policy</b></a></SmallText></SmallPrint>
        </Wrapper>
    )
}

export default Footer
