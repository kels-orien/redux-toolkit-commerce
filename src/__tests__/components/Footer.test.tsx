import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../../components/Footer'
import { BrowserRouter } from 'react-router-dom'

test('Footer render correctly', () => {
    const tree = renderer.create(<BrowserRouter><Footer/></BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot()

})