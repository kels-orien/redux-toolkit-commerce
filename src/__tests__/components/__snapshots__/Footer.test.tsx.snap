// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render correctly 1`] = `
<footer
  className="sc-bdnxRM dYdqok"
>
  <div
    className="sc-gtsrHT gXAmjl"
  >
    <div
      className="sc-dlnjwi jfQrVR"
    >
      <div
        className="sc-hKFxyN kJNMHm"
      >
        <div
          className="sc-jSFjdj eVbjtC"
        >
          <div
            className="sc-eCApnc KNIwK"
          >
            USEFUL LINKS
          </div>
          <div
            className="sc-gKAaRy cANxWp"
          >
            <ul>
              <a
                href="/products/category/electronics"
                onClick={[Function]}
              >
                <li>
                  Electronics
                </li>
              </a>
              <a
                href="/products/category/jewelery"
                onClick={[Function]}
              >
                <li>
                  Jewelry
                </li>
              </a>
              <a
                href="/products/category/men's clothing"
                onClick={[Function]}
              >
                <li>
                  Men's Clothing
                </li>
              </a>
              <a
                href="/products/category/women's clothing"
                onClick={[Function]}
              >
                <li>
                  Women's Clothing
                </li>
              </a>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="sc-hKFxyN kJNMHm"
      >
        <div
          className="sc-jSFjdj eVbjtC"
        >
          <div
            className="sc-eCApnc KNIwK"
          >
            HELP
          </div>
          <div
            className="sc-gKAaRy cANxWp"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  FAQ
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Shipping
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Payment
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Returns
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="sc-hKFxyN kJNMHm"
      >
        <div
          className="sc-jSFjdj eVbjtC"
        >
          <div
            className="sc-eCApnc KNIwK"
          >
            SOCIAL MEDIA
          </div>
          <div
            className="sc-gKAaRy cANxWp"
          >
            <ul>
              <li>
                <a
                  href="/"
                >
                  Facebook
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Instagram
                </a>
              </li>
              <li>
                <a
                  href="/"
                >
                  Twitter
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-iCoGMd hqNTnQ"
  >
    <div
      className="sc-pNWdM fCsxSp"
    >
      Â© 2021 Korien. All Rights Reserved. |
      <a
        href="/"
      >
        <b>
          Term of use
        </b>
      </a>
       | 
      <a
        href="/"
      >
        <b>
          Privacy Policy
        </b>
      </a>
    </div>
  </div>
</footer>
`;
